### DISCUSSION TITLE
How to use with uploaded images?

### DISCUSSION THREAD

**Discussion by mgsmus** [spatie/image#270] [Q&A]
Hello,

I was using v2 in my Laravel project and I could manipulate images before saving them like this:

```php
$avatar = $request->file('avatar')
    ->getRealPath();

Image::load($avatar)
    ->fit(Manipulations::FIT_CROP, 200, 200)
    ->save();

$path = Storage::putFile('avatars', $avatar); // S3
```

But now I have updated the package to v3 and modified my code as follows:

```php
Image::load($avatar)
    ->fit(Fit::Crop, 200, 200)
    ->save();
```

However, I am getting the following error:

```
Spatie\Image\Exceptions\UnsupportedImageFormat

Unsupported format ``.
```

For now, I have managed it with intervention/image-laravel like this:

```php
Image::read($avatar)
    ->cover(200, 200)
    ->save();
```

However, I would prefer to do this with spatie/image. Am I missing something?

Thank you.

**Comment by kikky7** [comment]
Same problem here with version `3.7.1`.

Unsupported format `` error. on save().

```
$convertedImage = Image::load($isUploadedFile ? $fileOrPath->path() : $fileOrPath);

$convertedImage
    ->fit(Fit::Crop, 150, 150)
    ->optimize()    
    ->save();
```

It works fine on older version `2.2.5`

```
$convertedImage = Image::load($isUploadedFile ? $fileOrPath->path() : $fileOrPath);

$convertedImage
    ->fit(Manipulations::FIT_CROP, 150, 150)
    ->optimize()
    ->save();
```

**Comment by timvandijck** [comment]
@mgsmus @kikky7 hey guys, I believe this error is caused because `getRealPath()` returns a path without a file extension (for example: `/private/var/tmp/phpfMBaek`).

In v3 we rely on a file extension to determine the file format we will be using to store the image.

In your case the solution would be to pass a file name with extension to the `save()` method:

```php
Image::loadFile($avatar)
        ->fit(Fit::Crop, 200, 200)
        ->save('avatar.png');
```

Ofcourse you can use your own logic to generate a unique filname.

**Reply by kikky7** [reply]
Yes, I realized that, but what I tried to achive is to resize the image "on the fly" before sending it to media library package because in this specific case I don't need an original file, without creating conversion (which saves original and converted file). With older version it worked, but now I just used the conversion instead.

So, in short, I want media library to handle the file saving, but to avoid creating conversions, since I don't need the original file, just the converted one.

**Reply by timvandijck** [reply]
You can just load the file from the request, the important thing is to just pass a filename with extension in the `save` method, the load part of your example should be fine.
